packages/accounts-controller/src/AccountsController.ts

```diff
@@ -682,19 +684,35 @@ export class AccountsController extends BaseController<
    */
   async #listNormalAccounts(): Promise<InternalAccount[]> {
     const internalAccounts: InternalAccount[] = [];
-    const { keyrings } = await this.messagingSystem.call(
+    const { keyrings, keyringsMetadata } = await this.messagingSystem.call(
       'KeyringController:getState',
     );
-    for (const keyring of keyrings) {
+
+    for (let keyringIndex = 0; keyringIndex < keyrings.length; keyringIndex++) {
+      const keyring = keyrings[keyringIndex];
       const keyringType = keyring.type;
       if (!isNormalKeyringType(keyringType as KeyringTypes)) {
         // We only consider "normal accounts" here, so keep looping
         continue;
       }
 
-      for (const address of keyring.accounts) {
+      for (
+        let accountIndex = 0;
+        accountIndex < keyring.accounts.length;
+        accountIndex++
+      ) {
+        const address = keyring.accounts[accountIndex];
         const id = getUUIDFromAddressOfNormalAccount(address);
 
+        let options = {};
+
+        if (isHdKeyringType(keyring.type as KeyringTypes)) {
+          options = {
+            entropySource: keyringsMetadata[keyringIndex].id,
+            derivationPath: getDerivationPathForIndex(accountIndex),
```

Comment:
As discussed, I think it would be great to have this one if you're ok?

```suggestion
            // NOTE: We are not using the `hdPath` from the associated keyring here and
            // getting the keyring instance here feels a bit overkill.
            // This will be naturally fixed once every keyring start using `KeyringAccount` and implement the keyring API.
            derivationPath: getDerivationPathForIndex(accountIndex),
```